### ISSUE TITLE
uncaught exception : object has been destroyed

### ISSUE THREAD

**Issue by kapsr** [particl/particl-desktop#305]
I now sometimes get this error when closing the GUI:

![5303d58a4d73e6d42308562f8abca10f](https://user-images.githubusercontent.com/32437236/32629691-54bcb360-c59a-11e7-8444-d6fc865bac7d.png)

**Comment by SativaL** [comment]
for me i got the same error in the final release.

Linux mint 18.2

> 
> Uncaught Exception:
> Error: Could not connect to daemon.
>     at App.<anonymous> (/home/net/Downloads/particl-1.0.0/resources/app.asar/modules/rpc/daemon.js:169:11)
>     at emitTwo (events.js:111:20)
>     at App.emit (events.js:194:7)
>     at Timeout.daemonStartup [as _onTimeout] (/home/net/Downloads/particl-1.0.0/resources/app.asar/modules/rpc/daemon.js:70:20)
>     at ontimeout (timers.js:386:14)
>     at tryOnTimeout (timers.js:250:5)
>     at Timer.listOnTimeout (timers.js:214:5)

**Comment by pciavald** [comment]
@kapsr can you enlarge the window so I can see full error on first line ?
@SativaL can you reproduce by launching from the command line and pasting the output ?

these are 2 distinct errors.

**Comment by pciavald** [comment]
@SativaL this is not the same issue. Please log it in #343. When doing so, can you please include the console logging, ie by launching ./partcl from the command line ?

@kapsr can you widen the error screen so we can read the full line ?

**Comment by pciavald** [comment]
seems observer was destroyed when called here
https://github.com/particl/partgui/blob/master/modules/rpc/rpc.js#L120

**Comment by pciavald** [comment]
maybe also involves a timeout
https://github.com/particl/partgui/blob/master/modules/rpc/rpc.js#L72

**Comment by pciavald** [comment]
also happened in Linux

```
Uncaught Exception:
Error: Object has been destroyed
at WebContents.send (/opt/Particl Wallet/resources/electron.asar/browser/api/web-contents.js:95:15)
at SafeSubscriber.observable.subscribe [as _error] (/opt/Particl Wallet/resources/app.asar/node_modules/rx-ipc-electron/lib/rx-ipc.js:48:25)
at SafeSubscriber.__tryOrUnsub (/opt/Particl Wallet/resources/app.asar/node_modules/rxjs/Subscriber.js:238:16)
at SafeSubscriber.error (/opt/Particl Wallet/resources/app.asar/node_modules/rxjs/Subscriber.js:197:26)
at Subscriber._error (/opt/Particl Wallet/resources/app.asar/node_modules/rxjs/Subscriber.js:128:26)
at Subscriber.error (/opt/Particl Wallet/resources/app.asar/node_modules/rxjs/Subscriber.js:102:18)
at rpcCall (/opt/Particl Wallet/resources/app.asar/modules/rpc/rpc.js:120:22)
at ClientRequest.request.on.error (/opt/Particl Wallet/resources/app.asar/modules/rpc/rpc.js:72:9)
at emitOne (events.js:96:13)
at ClientRequest.emit (events.js:191:7)
at Socket.socketCloseListener (_http_client.js:334:9)
at emitOne (events.js:101:20)
at Socket.emit (events.js:191:7)
at TCP._handle.close [as _onclose] (net.js:510:12)
```

**Comment by pciavald** [comment]
this should be fixed by #455. Please reopen if not fixed after Wallet 1.1.0 release.

**Comment by pciavald** [comment]
had this again :( 

```
Uncaught Exception:
Error: Object has been destroyed
    at WebContents.send (/home/pciavald/Documents/partgui/node_modules/electron/dist/resources/electron.asar/browser/api/web-contents.js:95:15)
    at SafeSubscriber.observable.subscribe [as _next] (/home/pciavald/Documents/partgui/node_modules/rx-ipc-electron/lib/rx-ipc.js:46:25)
    at SafeSubscriber.__tryOrUnsub (/home/pciavald/Documents/partgui/node_modules/rxjs/Subscriber.js:238:16)
    at SafeSubscriber.next (/home/pciavald/Documents/partgui/node_modules/rxjs/Subscriber.js:185:22)
    at Subscriber._next (/home/pciavald/Documents/partgui/node_modules/rxjs/Subscriber.js:125:26)
    at Subscriber.next (/home/pciavald/Documents/partgui/node_modules/rxjs/Subscriber.js:89:18)
    at rpc.call (/home/pciavald/Documents/partgui/modules/ipc/ipc.js:68:52)
    at IncomingMessage.response.on (/home/pciavald/Documents/partgui/modules/rpc/rpc.js:88:7)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:188:7)
    at endReadableNT (_stream_readable.js:975:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)

```

**Comment by kewde** [comment]
Fixed by #731

**Comment by pciavald** [comment]
just seen it on the discord app :O 
<img width="424" alt="capture_decran_2018-02-16_a_13 04 52" src="https://user-images.githubusercontent.com/3384097/36307154-14a74280-131b-11e8-9051-5127236a645f.png">
